scripts/lint-sh.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+find . -not \( -path './*.git/*' -prune \) -not \( -path './package/*' -prune \) -not \( -path './scripts/*' -prune \) -type f -exec bash -c 'if echo $1 | grep -qE ".*\.sh$" || head -n1 $1 | grep -qE "^#.*(sh|bash|dash|ksh)$"; then shellcheck -f gcc $1; fi' _ {} \;
```

Comment:
What is this line supposed to match, do we have anything to check outside of `package` and `scripts`? It seems to match the whole of `packages` - including the OpenWrt feeds...

Reply:
.git, package, scripts are excluded from that `find` call. Any other files (like contrib/*) will be checked. Yes openwrt needs also be excluded. I have fix it.

Reply:
Then let's just write "contrib"? Maybe even add that to the "scripts" find call?

Reply:
But if some scripts got added in the future in some other directory there will be no default check.
"scripts" has differed shellcheck parameters then "contrib".

Reply:
- Lint commands will always need to be adjusted for future changes - either by additional inclusions, or additional exclusions. Just listing the directories to include in the linting makes the code simpler, so I'd prefer to go with that.
- Does scripts *need* different parameters, or can we fix the scripts to follow the same guidelines?

Reply:
>     * Lint commands will always need to be adjusted for future changes - either by additional inclusions, or additional exclusions. Just listing the directories to include in the linting makes the code simpler, so I'd prefer to go with that.

I,d suggest to have an opt out solution, so if some MRs contains new scripts in different directory's within in Gluon the CI will check them automatically.

>     * Does scripts _need_ different parameters, or can we fix the scripts to follow the same guidelines?

I think so, e.g. their files like `scripts/modules.sh` or `scripts/default_feeds.sh` which does not have a shebang.

these messages comes from the scripts folder running the check parameters of the contrib folder:

```
scripts/sha512sum.sh:18:3: note: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?. [SC2181]
scripts/features.sh:20:4: warning: Can't follow non-constant source. Use a directive to specify location. [SC1090]
scripts/features.sh:51:8: warning: Declare and assign separately to avoid masking return values. [SC2155]
scripts/features.sh:61:8: warning: Declare and assign separately to avoid masking return values. [SC2155]
scripts/features.sh:76:4: warning: Can't follow non-constant source. Use a directive to specify location. [SC1090]
scripts/default_feeds.sh:1:1: error: Tips depend on target shell and yours is unknown. Add a shebang. [SC2148]
scripts/default_feeds.sh:1:1: warning: DEFAULT_FEEDS appears unused. Verify use (or export if used externally). [SC2034]
scripts/sha256sum.sh:18:3: note: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?. [SC2181]
scripts/modules.sh:1:1: error: Tips depend on target shell and yours is unknown. Add a shebang. [SC2148]
scripts/modules.sh:2:40: warning: Can't follow non-constant source. Use a directive to specify location. [SC1090]
scripts/update.sh:21:27: warning: commit is referenced but not assigned. [SC2154]
scripts/update.sh:22:14: warning: repo is referenced but not assigned. [SC2154]
scripts/update.sh:22:22: warning: branch is referenced but not assigned. [SC2154]
```

Reply:
> I,d suggest to have an opt out solution, so if some MRs contains new scripts in different directory's within in Gluon the CI will check them automatically.

Making the code more complex for an unimportant case that is unlikely to happen any time soon does not make sense to me.

Reply:
ok, changed.